sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -784,19 +785,19 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
       case _: BroadcastHashJoinExec => ()
     }.nonEmpty)
 
-    // Even with hint broadcast, hash join is not used for non-binary collated strings.
+    // Hash join is also used for non-binary collated strings.
```

Comment:
I think that you should rename the test. It's currently called `test("hash based joins not allowed for non-binary collated strings")` which is no longer true.

Reply:
We should assert that collationkey was injected into physical plan.